### ISSUE TITLE
[KitKat] AdAway won't start webserver but it can be started manualy through the terminal.

### ISSUE THREAD

**Issue by JORGETECH** [AdAway/AdAway#808]
When I activate the webserver option on the AdAway app the webserver is not started, the button is actually deactivated when I exit the app and open it again.

Trying to find the cause of the issue I activated AdAway logging and using the app CatLog I found that the command that AdAway uses for activating the webserver is: /data/app-lib/org.adaway-1/libblank_webserver_exec.so  > /dev/null 2>&1 & 2>&1

I decided to try that command under Terminal with superuser privileges and I found out that when the command is entered it outputs this to the terminal: [1] 18447. I also found that once that command is entered the webserver button is enabled on the AdAway app, so this seems to be an app problem.

The log I "recorded" using CatLog: https://drive.google.com/open?id=0B54QZ6UtmdtANDZnSzhjcDNUbmc

[Device info]
Device: Huawei Ascend G6 (U10)
Android version: 4.4.2 KitKat (Huawei EMUI 3.0 stock rom)
Busybox: 1.24.2-meefik
SU Binary: KingRoot 3.4.0 (?)

**Comment by sanjay900** [comment]
How is it with the latest version of adaway?

**Comment by JORGETECH** [comment]
If you mean version 3.1.2, that was the version I was testing with. Anyways I managed to make an init.d script that starts the webserver on boot, but this should be investigated for convenience.

**Comment by sanjay900** [comment]
I was thinking of setting if it happened with the 3.2 preview

**Comment by JORGETECH** [comment]
I don't know where to find that version. Can you send me a link? I also found that TCPdump does not start in a similar case of the webserver.

**Comment by sanjay900** [comment]
https://www.androidfilehost.com/?fid=312978532265366394

**Comment by JORGETECH** [comment]
Thanks, I'll try if it works with that version and report back.

**Comment by JORGETECH** [comment]
I just tried with the latest dev build, the same happens.

**Comment by nicman23** [comment]
Same happened with me, because the author of the ROM I m using, had AdAway preinstalled as a system app. Remounted system, deleted /system/app/Adaway , formatted davik reinstalled app and it now works

**Comment by JORGETECH** [comment]
So if I wipe the dalvik cache and reinstall it may work?

**Comment by nicman23** [comment]
that worked for me, but first remove the apk/ folder from the system partition (inside the app folder for me)

**Comment by JORGETECH** [comment]
I don't think that is the issue exactly. I tried and it happens after a fresh install. I am NOT using a custom ROM.
 This has to be something more technical, the command is executed by the program but nothing happens but if I try the command (that the app sends) through Terminal Emulator it actually works. So it seems it is some issue coming from the app itself.

**Comment by Sopor-** [comment]
I'm running NOX and the web server won't auto start. This is Android 4.4.2 and AdAway 3.2 (60).
